app/hooks/use-prepare-ledger.ts

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from 'react';
+import { LedgerError } from '@zondax/ledger-blockstack';
+import { Observable } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+import type { LedgerMessageEvents } from '../main/register-ledger-listeners';
+import { useListenLedger } from './use-listen-ledger';
+import { messages$ } from './use-message-events';
+
+export enum LedgerConnectStep {
+  Disconnected,
+  ConnectedAppClosed,
+  ConnectedAppOpen,
+  ActionComplete,
+}
+
+const ledgerEvents$ = messages$.pipe(
+  filter(value => value.type === 'ledger-event')
+) as Observable<LedgerMessageEvents>;
+
+export function usePrepareLedger() {
+  const [step, setStep] = useState<LedgerConnectStep>(LedgerConnectStep.Disconnected);
+  const [isLocked, setIsLocked] = useState(false);
+
+  useListenLedger();
+
+  useEffect(() => {
```

Comment:
could be its own hook?

Reply:
Not sure this is needed, other than including `useListenLedgerEffect` it already is, and better to encapsulate that here was it will always be needed with this one. Open to objections though.